### ISSUE TITLE
ethernet flow-control config

### ISSUE THREAD

**Issue by LimeHat** [openconfig/public#1044]
https://github.com/openconfig/public/blob/f46ca2cc6a2e5832cb0bd03d30c78b735129063c/release/models/interfaces/openconfig-if-ethernet.yang#L368-L382

Currently, `enable-flow-control` leaf does not specify in which direction the config is applied (tx, rx, or both).

It is common for implementations to support rx/tx granularity, e.g.:

- Arista [ref](https://www.arista.com/en/um-eos/eos-ethernet-ports#flowcontrol_receive)
```
flowcontrol receive STATE
flowcontrol send STATE
```
- Cisco [ref](https://www.cisco.com/c/en/us/td/docs/switches/datacenter/nexus3548/sw/qos/602_A1_1/b_N3548_QoS_Config_602_A11/b_N3548_QoS_Config_602_A11_chapter_0100.pdf)
```
# flowcontrol send on
# flowcontrol receive on
```
- Nokia [ref](https://documentation.nokia.com/srlinux/23-10/books/data-model-reference/srl_nokia-interfaces_0.html#d940)
```
usage: flow-control
  receive*          When this is true PAUSE frames received on this interface are accepted and processed
  transmit*         Enables/disables the sending of PAUSE frames.
```

**Comment by dplore** [comment]
Agreed, there is precedent to have asymmetric flow control.  FYI @atmanmehta (who investigating adding additional support for PFC)

**Comment by atmanmehta** [comment]
> Agreed, there is precedent to have asymmetric flow control. FYI @atmanmehta (who investigating adding additional support for PFC)

Ack. Thanks a lot for the confirmation

**Comment by iitrgaurav** [comment]
- Juniper [ref](https://www.juniper.net/documentation/us/en/software/junos/traffic-mgmt-qfx/topics/task/cos-asymmetric-flow-control-qfx-series-cli.html)
 
```
usage: configured-flow-control
set configured-flow-control rx-buffers (on | off) tx-buffers (on | off) 

```

**Comment by LimeHat** [comment]
i'd like to keep this open
